Add hooks to csv driver

winfried-ripken opened the pull request:
# Description

Add hooks to csv driver to include simple preprocessing steps for quick loading of small csv datasets

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring including code style reformatting 
- [ ] Other (please describe):

# Checklist:

- [ ] I have read the [contributing guideline doc](https://squirrel-core.readthedocs.io/en/latest/usage/code_of_conduct.html) (external contributors only)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have kept the PR small so that it can be easily reviewed  
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] All dependency changes have been reflected in the pip requirement files.

AlirezaSohofi commented on squirrel/driver/csv_driver.py:
```diff
@@ -14,14 +14,16 @@ class CsvDriver(FileDriver, DataFrameDriver):
 
         Args:
             path (str): Path to a .csv file.
+            df_hooks (Iterable[Callable], optional): preprocessing hooks to execute on the dataframe
```
Please mention that the first hook should accepts a single argument of type `dask.dataframe.DataFrame`. If that's not desirable, e.g. we want the hook to operate on pd.DataFrame, this need to be implemented too.

AlirezaSohofi reviewed (changes requested):
LG. Just a note on the docstring.

AlirezaSohofi commented:
@winfried-ripken  note this [comment](https://github.com/merantix-momentum/squirrel-datasets-core/pull/56#discussion_r982133207) from @jotterbach. Would make sense to give options to users to choose dask or pandas.

winfried-ripken commented:
> @winfried-ripken note this [comment](https://github.com/merantix-momentum/squirrel-datasets-core/pull/56#discussion_r982133207) from @jotterbach. Would make sense to give options to users to choose dask or pandas.

Thanks for pointing this out!
I have added a flag to control this and changed the docstring where it was necessary :)

AlirezaSohofi reviewed (approved):
:+1:
